Fix: Inserter impossible to collapse panels while searching.

jorgefilipecosta opened the pull request:
## Description
The inserter logic has a problem. If we open the inserter search for something e.g: 'c' that returns many categories and we try to collapse a panel nothing happens.
This PR fixes this problem.
The behavior of opening all panels each time the search string changes was kept.

## How has this been tested?
Add a paragraph type something.
Go to the inserter, write a "c" on the search.
Navigate on the inserter category panel and verify all panel are open, but it is possible to close them and open them again without problems.

jorgefilipecosta committed:
Fix: Inserter impossible to collapse panels while searching.

mapk commented:
I tried testing this and got several errors in console and Gutenberg just wasn't coming up for me.

jorgefilipecosta commented:
Hi @mapk thank you for trying to execute this PR. It is old so maybe we removed some dependency meanwhile that you already don't have on your machine. Anyway, I rebased this PR if some problem persists please run `npm install` before `npm run dev`, but I expect everything will run smoothly after this rebase. Sorry for the trouble cause.

gwwar commented on packages/editor/src/components/inserter/menu.js:
```diff
@@ -198,11 +198,16 @@ export class InserterMenu extends Component {
+					openPanels = openPanels.concat(
+						Object.keys( itemsPerCategory )
+					);
+				}
```
I think it might be worthwhile to break this out to it's own helper to help read what's happening here. For example:

```jsx
filterOpenPanels: ( filterValue, itemsPerCategory ) => {
    if( filterValue === this.state.filterValue ) {
        return this.state.openPanels;
    }
    if ( ! filterValue ) {
        return [ 'suggested' ];
    }
    if ( reusableItems.length > 0 ) {
        return [ 'reusable' ];
    }
    if( filteredItems.length > 0) {
        return [ ...Object.keys( itemsPerCategory ) ]
    }
    return [];
}
// in filter...
this.setState( {
   openPanels: this.filterOpenPanels( filterValue, itemsPerCategory ), 
   //...
} );
```

gwwar reviewed (approved):
Thanks @jorgefilipecosta this tests well for me! I left an optional note about code readability.

Before:
![before](https://user-images.githubusercontent.com/1270189/54075312-20b99080-425b-11e9-9d84-67338f83a1dc.gif)

After:
![after](https://user-images.githubusercontent.com/1270189/54075313-24e5ae00-425b-11e9-9ebe-8d431a27bde4.gif)

jorgefilipecosta committed:
Extract filterOpenPanels funcion

jorgefilipecosta commented:
Thank you for the review @gwwar 👍 Your feedback was addressed before the merge.
